File: spring/src/main/java/com/example/alten/product/dto/ProductResponseDto.java
package com.example.alten.product.dto;

import com.example.alten.category.dto.CategoryResponseDto;
import com.example.alten.inventoryStatus.dto.InventoryStatusResponseDto;
import com.example.alten.product.model.Product;

public class ProductResponseDto {

    public Long id;
    public String code;
    public String name;
    public String description;
    public int price;
    public int quantity;
    public InventoryStatusResponseDto inventoryStatus;
    public CategoryResponseDto category;
    public String image;
    public Integer rating;

    public ProductResponseDto(Product product) {
        this.id = product.getId();
        this.code = product.getCode();
        this.name = product.getName();
        this.description = product.getDescription();
        this.price = product.getPrice();
        this.quantity = product.getQuantity();
        this.inventoryStatus = new InventoryStatusResponseDto(product.getInventoryStatus());
        this.category = new CategoryResponseDto(product.getCategory());
        this.image = product.getImage();
        this.rating = product.getRating();
    }
}

File: nest/src/product/dto/product-response.dto.ts
import { CURRENCY_TO_CENTS } from 'src/shared/helpers/number.helper';
import { IProduct } from '../interface/product.entity';

export class ProductResponseDto {
    id: number;
    code: string;
    name: string;
    description: string;
    price: number;
    quantity: number;
    inventoryStatus: { id: number; label: string };
    category: { id: number; label: string };
    image: string | null;
    rating: number | null;

    constructor(product: IProduct) {
        this.id = product.id;
        this.code = product.code;
        this.name = product.name;
        this.description = product.description;
        this.price = product.price / CURRENCY_TO_CENTS;
        this.quantity = product.quantity;
        this.inventoryStatus = {
            id: product.inventoryStatus.id,
            label: product.inventoryStatus.label,
        };
        this.category = {
            id: product.category.id,
            label: product.category.label,
        };
        this.image = product.image;
        this.rating = product.rating;
    }
}
